tests/suites/caas_filesystem/import.sh

```diff
@@ -0,0 +1,52 @@
+test_import_filesystem() {
+        # Echo out to ensure nice output to the test suite.
+        echo
+
+        # Ensure a bootstrap Juju model exists.
+        model_name="import-filesystem"
+        file="${TEST_DIR}/test-${model_name}.log"
+        ensure "${model_name}" "${file}"
+
+        # Create a PersistentVolume by deploying and deleting an application.
+        echo "Create persistent volume to be imported"
+        juju deploy postgresql-k8s --channel 14/stable --trust
+        wait_for "postgresql-k8s" "$(active_idle_condition "postgresql-k8s" 0)"
+
+        # Capture the provisioned PersistentVolume ID.
+        PV=$(juju storage --format json | jq -r '.volumes["0"]."provider-id"')
+
+        # Clean up: remove the application and associated storage (retain PV).
+        juju remove-application postgresql-k8s --no-prompt
+        wait_for "{}" ".applications"
+        juju remove-storage pgdata/0 --no-destroy
+        wait_for "{}" ".storage"
+
+        # Attempt to import the PersistentVolume: expect failure due to reclaim policy.
+        set +e
+        OUT=$(juju import-filesystem kubernetes "${PV}" pgdata 2>&1)
+        set -e
+        echo "${OUT}" | check \
+                "importing volume \"${PV}\" with reclaim policy \"Delete\" not supported \(must be \"Retain\"\)"
+
+        # Fix: update the PersistentVolume's reclaim policy to 'Retain'.
+        kubectl patch pv "${PV}" -p '{"spec":{"persistentVolumeReclaimPolicy":"Retain"}}'
+
+        # Attempt to import the PersistentVolume: expect failure due to existing claimRef.
+        set +e
+        OUT=$(juju import-filesystem kubernetes "${PV}" pgdata 2>&1)
+        set -e
+        echo "${OUT}" | check \
+                "importing volume \"${PV}\" already bound to a claim not supported"
+
+        # Fix: remove the claimRef from the PersistentVolume.
+        kubectl patch pv "${PV}" --type merge -p '{"spec":{"claimRef": null}}'
```

Comment:
Running QA fails for me because after this command, the claimRef is not set to null - well it is, but it flips back.
The model namespace still contains the PVC that the PV is bound to. Deleting this PVC allows the claimRef to stay as null when set.

Reply:
Hum...
This works for me in the local. But you are right, delete the pvc is a more proper way.